### PR TITLE
Bugfix for Eichrecht Signed Data

### PR DESCRIPTION
**Pull request by WillyLellouch**

### CODE REVIEW

File: src/server/ocpp/utils/OCPPUtils.ts [hunk 1]
```diff
@@ -692,6 +692,24 @@ export default class OCPPUtils {
 
   public static updateTransactionWithStopTransaction(transaction: Transaction, stopTransaction: OCPPStopTransactionRequestExtended,
       user: User, alternateUser: User, tagId: string): void {
+    // Handle Signed Data
+    for(const meterValue of (stopTransaction.transactionData as OCPPMeterValue[])){
+      if(meterValue.sampledValue){
```
Comment (jerome-benoit) [inline src/server/ocpp/utils/OCPPUtils.ts:697]: You development setup does not have the mandatory vscode extensions installed and allowed: that PR is very badly formatted.

File: src/server/ocpp/utils/OCPPUtils.ts [hunk 2]
```diff
@@ -692,6 +692,24 @@ export default class OCPPUtils {
...
+        for(const sampledValue of meterValue.sampledValue){
+          if (sampledValue.format === OCPPValueFormat.SIGNED_DATA) {
+            if (sampledValue.context === OCPPReadingContext.TRANSACTION_BEGIN) {
+              // Set the first Signed Data and keep it
+              transaction.signedData = sampledValue.value as string;
+              continue;
+            } else if (sampledValue.context === OCPPReadingContext.TRANSACTION_END) {
+              // Set the last Signed Data (used in the last consumption)
+              transaction.currentSignedData = sampledValue.value as string;
+              continue;
```
Comment (jerome-benoit) [inline src/server/ocpp/utils/OCPPUtils.ts:707]: I do not understand why you need to specify a continue after each findings. 

And that code duplicate a similar code in `updateTransactionWithMeterValues()`. Please factor out the code digging the metervalues.

**Review by jerome-benoit — CHANGES_REQUESTED** [review]
Deadline to properly implement that fix with unit tests fully running is tomorrow.

File: src/server/ocpp/utils/OCPPUtils.ts (hunk 1 above)
Comment (WillyLellouch) [reply src/server/ocpp/utils/OCPPUtils.ts]: reformated

File: src/server/ocpp/utils/OCPPUtils.ts (hunk 2 above)
Comment (WillyLellouch) [reply src/server/ocpp/utils/OCPPUtils.ts]: the code is factored out in OCPPUtils

**Comment by jerome-benoit** [comment]
Could you please provide in the description an explaination on the changes made over time? 
We need to understand why it took so long to fix the issue.

**Review by jerome-benoit — APPROVED** [review]

**Review by jerome-benoit — APPROVED** [review]

**Pull request merged**